lib/vnet/network_stack_darwin.go

```diff
@@ -0,0 +1,22 @@
+// Teleport
+// Copyright (C) 2025 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package vnet
+
+type NetworkStackInfo struct {
+	// ifaceName is the name of the interface used by VNet, "utun" followed by a number.
+	IfaceName string
+}
```

Comment:
can we actually just make this platform-independent and include a godoc?
on windows we can control the interface name and it's a constant, you can just set it to `tunInterfaceName`